File: 2022/go/2.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strings"
)

func main() {

	input, err := os.Open("../2.txt")
	if err != nil {
		fmt.Printf("Could not read the file due to this %s error \n", err)
	}
	sc := bufio.NewScanner(input)

	responses := map[string]int{ "X": 1, "Y": 2, "Z": 3,}
	part1 := map[string]map[string]int{
		"A": { "X": 3, "Y": 6, "Z": 0, },
        "B": { "X": 0, "Y": 3, "Z": 6, },
        "C": { "X": 6, "Y": 0, "Z": 3, },
	}

	results := map[string]int{ "X": 0, "Y": 3, "Z": 6,}
	part2 := map[string]map[string]int{
		"A": { "X": 3, "Y": 1, "Z": 2, },
        "B": { "X": 1, "Y": 2, "Z": 3, },
        "C": { "X": 2, "Y": 3, "Z": 1, },
	}

	var score_part1 int = 0
	var score_part2 int = 0

	for sc.Scan() {
		if v := len(sc.Text()); v > 0 {
			line := strings.Split(sc.Text(), " ")
			opponent := line[0]
            response := line[1]

			score_part1 += responses[response]
			score_part2 += results[response]

			score_part1 += part1[opponent][response]
			score_part2 += part2[opponent][response]

		}
	}
	
	fmt.Printf("Score 1: %v\n", score_part1)	
	fmt.Printf("Score 2: %v\n", score_part2)	
}

File: 2022/rust/2.rs
use std::fs;
use std::collections::HashMap;

fn main() {
    let input = fs::read_to_string("../2.txt").unwrap();
    let lines : Vec<&str> = input.split("\n").collect();

    let mut responses = HashMap::from([("X", 1), ("Y", 2), ("Z", 3)]);
    let mut part1 = HashMap::from(
        [
            ("A", HashMap::from([("X", 3), ("Y", 6), ("Z", 0)])),
            ("B", HashMap::from([("X", 0), ("Y", 3), ("Z", 6)])),
            ("C", HashMap::from([("X", 6), ("Y", 0), ("Z", 3)])),
        ]
    );

    let mut results = HashMap::from([("X", 0), ("Y", 3), ("Z", 6)]);
    let mut part2 = HashMap::from(
        [
            ("A", HashMap::from([("X", 3), ("Y", 1), ("Z", 2)])),
            ("B", HashMap::from([("X", 1), ("Y", 2), ("Z", 3)])),
            ("C", HashMap::from([("X", 2), ("Y", 3), ("Z", 1)])),
        ]
    );

    let mut score_part1 : i32 = 0;
    let mut score_part2 : i32 = 0;
    for line in lines {
        let round : Vec<&str> = line.trim().split(" ").collect();
        if (line.len() > 0) {
            let opponent = round[0];
            let response = round[1];
            score_part1 += responses.get(&response).unwrap();
            score_part2 += results.get(&response).unwrap();
            
            
            let p1 = part1.get(&opponent).unwrap();
            let p2 = part2.get(&opponent).unwrap();
            score_part1 += p1.get(&response).unwrap();
            score_part2 += p2.get(&response).unwrap();
   
        }
    }
    
    println!("Score 1: {}", score_part1);
    println!("Score 2: {}", score_part2);
    
}

